### DISCUSSION TITLE
wireguard 能over udphop吗？

### DISCUSSION THREAD

**Discussion by marsgong** [cnbatch/udphop#9] [General]
我配置好了，拨号握手成功了，但在wireguard客户端出站带宽被占满。

**Comment by cnbatch** [comment]
可能是出现了路由环路。
在Wireguard连接建立完成后，需要给路由表加一个条目，目标服务器的IP地址继续走WAN口而不走VPN通道。
灵活一点的做法是，建立通道但不要把所有流量都交给wireguard。具体来说，AllowedIPs只填写隧道地址，最多加上内网地址，但不要写成0.0.0.0/0、::/0，然后再由防火墙（netfilter或iptables或ipfw或pf）做策略路由，而不是交给路由表去做。

**Comment by marsgong** [comment]
感谢回复！！
把客户端的wireguard的AllowedIPs 配置中把服务器的目标IP地址剔除就OK。
感谢。
原有UDP 转TCP性能影响太大，这解决了性能问题，再次感谢。
